Sort button to showcase if a column is sorted or not

mnmt7 opened the pull request:
closes #1244 

This PR adds a sort button which is visible only when the sort is applied on a column. Clicking on sort button opens a popover where the user can select other columns for sorting and as well as select the sorting direction i.e. ascending or descending.

Also on a side note, I noticed that even if you remove the sort, it is not removed from the Firebase and if you reload the page after removing the sort, it will be reapplied on the table. This should be an easy fix and I can do it in this PR if you say so.

/claim #1244

iamanishroy commented:
> Also on a side note, I noticed that even if you remove the sort, it is not removed from the Firebase and if you reload the page after removing the sort, it will be reapplied on the table. This should be an easy fix and I can do it in this PR if you say so.

Hey @mnmt7, it would be great if you could work on it. Also, please check the requested change as it is related to this fix.

mnmt7 commented:
Hey @iamanishroy, didn't get what you meant by the requested change?

harinij commented:
Hi @mnmt7 I think @iamanishroy is indicating that it would be great if you can also work on the following point you raised as part of this PR.

> Also on a side note, I noticed that even if you remove the sort, it is not removed from the Firebase and if you reload the page after removing the sort, it will be reapplied on the table. This should be an easy fix and I can do it in this PR if you say so.

mnmt7 commented:
@iamanishroy, I have implemented the requested changes Now, when a user removes the sort and reloads the page, the previous sort won't be reapplied. As per your suggestion, I have utilized the `triggerSaveTableSorts()` function to achieve this. However, I would like to confirm one thing. It seems to me that `triggerSaveTableSorts()` is intended to modify sort settings for individual users rather than applying them for all users. Please clarify if my understanding is correct.

The current flow of the sort is as follows:

1. The user applies the sort.
2. A popup appears, giving the option to save the sort for all users.
3. If the user chooses not to save the sort and:
    - The user removes the sort and refreshes the page, the sort won't be reapplied.
4. If the user chooses to save the sort and:
    - The user removes the sort, the changes are immediately reflected on the screen i.e. the sort is removed. However, if the page is reloaded, the sort will be reapplied because now the "sort settings for all users" will take the precedence. In short, if no user-specific sort setting exists, the sort settings for all users will be utilized.

Furthermore, there was one more thing I wanted to mention. Regarding the design of the columns dropdown, I made a slight modification. Instead of the design shown in the provided images, I have opted for a slightly different design:

Requested design -
![236119112-caed044e-1c03-4c25-b054-1428d97ff38d](https://github.com/rowyio/rowy/assets/107614965/effc57b8-652d-4817-b7a3-b4474ec5a3ad)

Implemented design -
![Screenshot 2023-06-01 at 10-16-30 Rowy – GCP as easy as ABC](https://github.com/rowyio/rowy/assets/107614965/b291ee30-d786-4ea0-a5d1-0356feff43f3)

I chose this design since the dropdown component was already being used for the filter functionality, and it seemed logical to maintain consistency. The only issue with this was what should be the action performed when the "clear" option is clicked. Right now, the sort is removed and consequently the sort modal disappears.

iamanishroy reviewed (approved):
LGTM
